### PR TITLE
Assembly devices minor refactor & code cleanup

### PR DESCRIPTION
**Pull request by Citinited**
List of code changes:

- Removes a tonne of useless brackets (I don't think I removed any vital ones, could be worth a look though)
- `return 1` and `return 0` are now `return TRUE` and `return FALSE`
- Puts a bunch of proc checks on new lines (i.e. 
```
if(X) return

goes to

if(X)
    return
```
- Adds helpers
- Eviscerates all colons from this code
- Goodbye implied vars, `as obj|mob` and `.0`
- Absolutely paths all devices

No CL, entirely backend

### CODE REVIEW

File: code/modules/assembly/bomb.dm [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
 		qdel(src)
 		return
-	if((istype(W, /obj/item/weldingtool) && W:welding))
+	if(istype(W, /obj/item/weldingtool) && W:welding)
```
Comment (Fox-McCloud) [inline code/modules/assembly/bomb.dm]: `W:welding`. 

😠

File: code/modules/assembly/holder.dm [hunk 2]
```diff
@@ -13,200 +13,201 @@
...
-		D:holder = src
-		D2:holder = src
-		a_left = D
-		a_right = D2
-		name = "[D.name]-[D2.name] assembly"
-		update_icon()
-		return 1
-
-
+/obj/item/assembly_holder/proc/attach(var/obj/item/D, var/obj/item/D2, var/mob/user)
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: no implied `var` for arguments.

File: code/modules/assembly/holder.dm [hunk 3]
```diff
@@ -13,200 +13,201 @@
...
+	if(a_right)
+		a_right.holder = null
+	return ..()
+
+/obj/item/assembly_holder/attach(var/obj/item/D, var/obj/item/D2, var/mob/user)
+	if(!D || !D2)
+		return FALSE
+	if(!isassembly(D) || !isassembly(D2))
+		return FALSE
+	if(D:secured || D2:secured)
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: good heavens, why.

File: code/modules/assembly/holder.dm [hunk 4]
```diff
@@ -13,200 +13,201 @@
...
+			overlays += "[O]_r"
+	if(master)
+		master.update_icon()
+
+
+/obj/item/assembly_holder/examine(mob/user)
+	..(user)
+	if(in_range(src, user) || loc == user)
+		if(secured)
+			to_chat(user, "\The [src] is ready!")
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: no `\The`, please.

File: code/modules/assembly/holder.dm [hunk 5]
```diff
@@ -13,200 +13,201 @@
...
+
+/obj/item/assembly_holder/attack_hand()//Perhapse this should be a holder_pickup proc instead, can add if needbe I guess
+	if(a_left && a_right)
+		a_left.holder_movement()
+		a_right.holder_movement()
+	..()
+	return
+
+/obj/item/assembly_holder/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(istype(W, /obj/item/screwdriver))
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: helper.

File: code/modules/assembly/holder.dm [hunk 6]
```diff
@@ -13,200 +13,201 @@
...
+		if(!a_left || !a_right)
+			to_chat(user, "<span class='warning'>BUG:Assembly part missing, please report this!</span>")
 			return
+		a_left.toggle_secure()
+		a_right.toggle_secure()
+		secured = !secured
+		if(secured)
+			to_chat(user, "<span class='notice'>\The [src] is ready!</span>")
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: beep.

File: code/modules/assembly/holder.dm [hunk 7]
```diff
@@ -13,200 +13,201 @@
...
-			else
-				a_left.attack_self(user)
-				a_right.attack_self(user)
+/obj/item/assembly_holder/attack_self(mob/user as mob)
+	add_fingerprint(user)
+	if(secured)
+		if(!a_left || !a_right)
+			to_chat(user, "<span class='warning'>Assembly part missing!</span>")
+			return
+		if(istype(a_left,a_right.type))//If they are the same type it causes issues due to window code
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: spacing.

File: code/modules/assembly/holder.dm [hunk 8]
```diff
@@ -13,200 +13,201 @@
...
+			switch(alert("Which side would you like to use?",,"Left","Right"))
+				if("Left")	a_left.attack_self(user)
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: new lines.

File: code/modules/assembly/holder.dm [hunk 9]
```diff
@@ -13,200 +13,201 @@
...
-				a_right.loc = T
-			spawn(0)
+			a_left.attack_self(user)
+			a_right.attack_self(user)
+	else
+		var/turf/T = get_turf(src)
+		if(!T)
+			return FALSE
+		if(a_left)
+			a_left:holder = null
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: AGHhhh. Why is assembly code so bad?!

File: code/modules/assembly/holder.dm [hunk 10]
```diff
@@ -13,200 +13,201 @@
...
+			a_left.loc = T
+		if(a_right)
+			a_right:holder = null
+			a_right.loc = T
+		spawn(0)
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: ....really.

File: code/modules/assembly/holder.dm [hunk 11]
```diff
@@ -13,200 +13,201 @@
...
+	if(master)
+		master.receive_signal()
+	return TRUE
+
+
+/obj/item/assembly_holder/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: not needed.

File: code/modules/assembly/holder.dm [hunk 12]
```diff
@@ -13,200 +13,201 @@
...
+		if(a_left != D)
+			a_left.pulsed(0)
+	if(master)
+		master.receive_signal()
+	return TRUE
+
+
+/obj/item/assembly_holder/ex_act(severity)
+	switch(severity)
+		if(1.0)
```
Comment (Fox-McCloud) [inline code/modules/assembly/holder.dm]: doesn't need `.0`

File: code/modules/assembly/igniter.dm [hunk 13]
```diff
@@ -21,21 +21,22 @@
...
 /obj/item/assembly/igniter/activate()
-	if(!..())	return 0//Cooldown check
+	if(!..())
+		return FALSE//Cooldown check
 	var/turf/location = get_turf(loc)
 	if(location)	location.hotspot_expose(1000,1000)
-	if(istype(src.loc,/obj/item/assembly_holder))
-		if(istype(src.loc.loc, /obj/structure/reagent_dispensers/fueltank/))
-			var/obj/structure/reagent_dispensers/fueltank/tank = src.loc.loc
+	if(istype(loc,/obj/item/assembly_holder))
```
Comment (Fox-McCloud) [inline code/modules/assembly/igniter.dm]: spacing.

File: code/modules/assembly/mousetrap.dm [hunk 14]
```diff
@@ -8,124 +8,118 @@
...
+				var/mob/living/carbon/human/user = usr
+				if((user.getBrainLoss() >= 60 || (CLUMSY in user.mutations)) && prob(50))
+					to_chat(user, "Your hand slips, setting off the trigger.")
+					pulse(0)
+		update_icon()
+		if(usr)
+			playsound(usr.loc, 'sound/weapons/handcuffs.ogg', 30, 1, -3)
+
+/obj/item/assembly/mousetrap/describe()
+	return "The pressure switch is [armed?"primed":"safe"]."
```
Comment (Fox-McCloud) [inline code/modules/assembly/mousetrap.dm]: spacing

File: code/modules/assembly/mousetrap.dm [hunk 15]
```diff
@@ -8,124 +8,118 @@
...
+					H.Stun(3)
+		if(affecting)
+			affecting.receive_damage(1, 0)
+	else if(ismouse(target))
+		var/mob/living/simple_animal/mouse/M = target
+		visible_message("<span class='danger'>SPLAT!</span>")
+		M.splat()
+	playsound(loc, 'sound/effects/snap.ogg', 50, 1)
+	layer = MOB_LAYER - 0.2
+	armed = 0
```
Comment (Fox-McCloud) [inline code/modules/assembly/mousetrap.dm]: `TRUE/FALSE`

File: code/modules/assembly/timer.dm [hunk 16]
```diff
@@ -14,127 +14,118 @@
...
+	timing = !timing
+	update_icon()
+	return FALSE
+
+/obj/item/assembly/timer/toggle_secure()
+	secured = !secured
+	if(secured)
+		processing_objects.Add(src)
+	else
+		timing = 0
```
Comment (Fox-McCloud) [inline code/modules/assembly/timer.dm]: beep.

File: code/modules/assembly/timer.dm [hunk 17]
```diff
@@ -14,127 +14,118 @@
...
+	return
+
+/obj/item/assembly/timer/process()
+	if(timing && (time > 0))
+		time -= 2 // 2 seconds per process()
+	if(timing && time <= 0)
+		timing = repeat
 		timer_end()
+		time = set_time
+	return
```
Comment (Fox-McCloud) [inline code/modules/assembly/timer.dm]: beep.

File: code/modules/assembly/timer.dm [hunk 18]
```diff
@@ -14,127 +14,118 @@
...
+
+/obj/item/assembly/timer/update_icon()
+	overlays.Cut()
+	attached_overlays = list()
+	if(timing)
+		overlays += "timer_timing"
+		attached_overlays += "timer_timing"
+	if(holder)
+		holder.update_icon()
+	return
```
Comment (Fox-McCloud) [inline code/modules/assembly/timer.dm]: beep.

File: code/modules/assembly/bomb.dm (hunk 1 above)
Comment (Citinited) [reply code/modules/assembly/bomb.dm]: I cringed when I saw how many colons were in this

File: code/modules/assembly/timer.dm [hunk 19]
```diff
@@ -14,127 +14,118 @@
...
+	</TT>
+	<BR><BR>
+	<A href='?src=[UID()];refresh=1'>Refresh</A>
+	<BR><BR>
+	<A href='?src=[UID()];close=1'>Close</A>"}
+	var/datum/browser/popup = new(user, "timer", name, 400, 400)
+	popup.set_content(dat)
+	popup.open(0)
+	onclose(user, "timer")
+	return
```
Comment (Fox-McCloud) [inline code/modules/assembly/timer.dm]: beep.

File: code/modules/assembly/timer.dm [hunk 20]
```diff
@@ -14,127 +14,118 @@
...
+
+/obj/item/assembly/timer/Topic(href, href_list)
+	..()
+	if(!usr.canmove || usr.stat || usr.restrained() || !in_range(loc, usr))
```
Comment (Fox-McCloud) [inline code/modules/assembly/timer.dm]: `incapacitated` check, instead

**Comment by Citinited** [comment]
@Fox-McCloud Thanks for review, I'm going to bed but I'll get your changes made on Saturday

File: code/modules/assembly/holder.dm [hunk 21]
```diff
@@ -13,200 +13,201 @@
...
+		return FALSE
+	if(!D.remove_item_from_storage(src))
+		if(user)
+			user.remove_from_mob(D)
+		D.loc = src
+	if(!D2.remove_item_from_storage(src))
+		if(user)
+			user.remove_from_mob(D2)
+		D2.loc = src
+	D:holder = src
```
Comment (Citinited) [inline code/modules/assembly/holder.dm]: Okay so I have no idea what this `holder` variable is referring to. Anyone know?

**Comment by Citinited** [comment]
@Fox-McCloud All your review points have been addressed, let me know if you can see anything else

**Pull request merged**